appservice/src/SiteWrapper.ts

```diff
@@ -529,32 +521,86 @@ export class SiteWrapper {
-        let deploymentId: string = 'latest';
-        let deployment: DeployResult = await kuduClient.deployment.getResult(deploymentId);
-        while (!deployment.complete) {
-            if (!deployment.isTemp && deployment.id) {
-                // Switch from 'latest' to the permanent/unique id as soon as it's available
-                deploymentId = deployment.id;
+    private async waitForDeploymentToComplete(kuduClient: KuduClient, outputChannel: vscode.OutputChannel, deploySendTime: Date, pollingInterval: number = 5000): Promise<void> {
+        const alreadyDisplayedLogs: string[] = [];
+        let nextTimeToDisplayWaitingLog: number = Date.now();
+        let permanentId: string | undefined;
+        let deployment: DeployResult | undefined;
+
+        // tslint:disable-next-line:no-constant-condition
+        while (true) {
+            if (permanentId) {
+                deployment = await kuduClient.deployment.getResult(permanentId);
+            } else {
+                // Unfortunately, Kudu doesn't provide a unique/permanent id for a deployment right after it's started
+                // Instead, search for the deployment that was received just after we started the deployment
+                deployment = (await kuduClient.deployment.getDeployResults())
+                    .filter((deployResult: DeployResult) => deployResult.receivedTime && deployResult.receivedTime > deploySendTime)
+                    .sort((a: DeployResult, b: DeployResult) => b.receivedTime.valueOf() - a.receivedTime.valueOf())
+                    .shift();
+                if (!deployment.isTemp) {
+                    // Once the deployment is permanent, we can use use that id rather than searching based on received time
+                    permanentId = deployment.id;
+                }
             }
 
-            if (deployment.progress) {
-                this.log(outputChannel, deployment.progress);
+            if (deployment === undefined || !deployment.id) {
+                throw new Error(localize('failedToFindDeployment', 'Failed to get status of deployment.'));
             }
 
-            await new Promise((resolve: () => void): void => { setTimeout(resolve, pollingInterval); });
-            deployment = await kuduClient.deployment.getResult(deploymentId);
-        }
-        if (previousDeploymentId !== undefined && deployment.id === previousDeploymentId) {
-            throw new Error(localize('unsuccessfulDeployment', 'The deployment was unsuccessful. If this is a Local Git deployment, this could be a duplicate deployment.'));
+            let logEntries: LogEntry[] = [];
+            try {
+                logEntries = <LogEntry[]>await kuduClient.deployment.getLogEntry(deployment.id);
+            } catch (error) {
+                // Swallow 404 errors for temp deployments
+                // (it's expected that the temp logs don't exist once the deployment switches to permanent)
```

Comment:
> // (it's expected that the temp logs don't exist once the deployment switches to permanent) 

I don't understand this comment. Are you saying that you're expecting 404 for permanent logs? I don't see that in the code.